app/helpers/application_helper.rb

```diff
@@ -28,7 +28,11 @@ def banned_banner
     message = ban_error_message(domains, valid_until)
 
     content_tag(:div, class: 'ui message ban') do
-      content_tag(:div, message, class: 'header')
+      result = content_tag(:div, message, class: 'header')
+      if domains && domains.count < Setting.ban_number_of_strikes
+        result << content_tag(:p, violation_message(domains.count))
+      end
+      result
```

Comment:
Any chance to avoid using helpers and using pure objects instead?

Reply:
I prefer not to - we are already in helper using helpers here, so I prefer to keep the style constant.